src/markup.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  ForceReply,
+  InlineKeyboardButton,
+  InlineKeyboardMarkup,
+  KeyboardButton,
+  ReplyKeyboardMarkup,
+  ReplyKeyboardRemove,
+} from 'typegram'
+import { is2D } from './core/helpers/check-utils'
+
+type Hideable<B> = B & { hide?: boolean }
+type HideableKBtn = Hideable<KeyboardButton>
+type HideableIKBtn = Hideable<InlineKeyboardButton>
+
+class Markup<
+  T extends
+    | InlineKeyboardMarkup
+    | ReplyKeyboardMarkup
+    | ReplyKeyboardRemove
+    | ForceReply
+> {
+  constructor(readonly reply_markup: T) {}
+
+  selective<T extends ForceReply | ReplyKeyboardMarkup>(
+    this: Markup<T>,
+    value = true
+  ) {
+    this.reply_markup.selective = value
+    return this
+  }
+
+  resize(this: Markup<ReplyKeyboardMarkup>, value = true) {
+    this.reply_markup.resize_keyboard = value
+    return this
+  }
+
+  oneTime(this: Markup<ReplyKeyboardMarkup>, value = true) {
+    this.reply_markup.one_time_keyboard = value
+    return this
+  }
+}
```

Comment:
I'm not certain, but I believe immutabel chaining is simpler, what do you think?
```suggestion
  selective<T extends ForceReply | ReplyKeyboardMarkup>(
    this: Markup<T>,
    value = true
  ) {
    return new Markup({ ...this.reply_markup, selective: value })
  }

  resize(this: Markup<ReplyKeyboardMarkup>, value = true) {
    return new Markup({ ...this.reply_markup, resize_keyboard: value })
  }

  oneTime(this: Markup<ReplyKeyboardMarkup>, value = true) {
    return new Markup({ ...this.reply_markup, one_time_keyboard: value })
  }
```

Reply:
I do not see how it's simpler on our end, but I generally like immutability. It allows for reusing one markup instance and making slightly different modifications to it when sending multiple messages. Mutability would cause weird behaviour, so I applied your suggestion in 855f480